cms/djangoapps/contentstore/rest_api/v0/views/tabs.py

```diff
@@ -0,0 +1,239 @@
+        ],
+        responses={
+            204: "In case of success, a 204 is returned with no content.",
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def post(self, request: Request, course_id: str) -> Response:
+        """
+        Change visibility of tabs in a course.
+
+        **Example Requests**
+
+        You can provide either a tab_id or a tab_location.
+
+        Hide a course tab using ``tab_id``:
+
+            POST /api/contentstore/v0/tabs/{course_id}/settings/?tab_id={tab_id} {
+                "is_hidden": true
+            }
+
+        Hide a course tab using ``tab_location``
+
+            POST /api/contentstore/v0/tabs/{course_id}/settings/?tab_location={tab_location} {
+                "is_hidden": true
+            }
+
+        **Response Values**
+
+        If the request is successful, an HTTP 204 response is returned
+        without any content.
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_write_access(request.user, course_key):
+            raise self.permission_denied(request)
+
+        tab_id_locator = TabIDLocatorSerializer(data=request.query_params)
+        tab_id_locator.is_valid(raise_exception=True)
+
+        course_module = modulestore().get_course(course_key)
+        serializer = CourseTabUpdateSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        edit_tab_handler(
+            course_module,
+            {
+                "tab_id_locator": tab_id_locator.data,
+                **serializer.data,
+            },
+            request.user,
+        )
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@view_auth_classes(is_authenticated=True)
+class CourseTabReorderView(DeveloperErrorViewMixin, APIView):
+    """
+    API view for reordering course tabs.
+    """
+
+    def handle_exception(self, exc: Exception) -> Response:
+        """
+        Handle NotImplementedError and return a proper response for it.
+        """
+        if isinstance(exc, NotImplementedError):
+            return self._make_error_response(400, str(exc))
+        return super().handle_exception(exc)
+
+    @apidocs.schema(
+        body=[TabIDLocatorSerializer],
+        parameters=[
+            apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID"),
+        ],
+        responses={
+            204: "In case of success, a 204 is returned with no content.",
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def post(self, request: Request, course_id: str) -> Response:
```

Comment:
why this reorder API is needed?

Reply:
In the old code there is a single API to both edit and reorder tabs. I've split that into two. 
The reorder endpoint allows changing the order of tabs.

Reply:
Ok, is it just to refactor some old code? I do not think the Pages & Resouces view allow / support this feature?

Reply:
It doesn't yet. However, eventually, it will in order to support reordering tabs when that feature needs to be added there.

Reply:
Sure, thank you. +1